### PR TITLE
Embedded DNS problem after renaming container. Step1:change in libnetwork side

### PR DESCRIPTION
**Pull request by jmzwcn**
Embedded DNS problem after renaming container. Step1:change in libnetwork side
#22466 
Signed-off-by: Daniel Zhang <jmzwcn@gmail.com>

### CODE REVIEW

File: endpoint.go [hunk 1]
```diff
@@ -533,13 +533,16 @@ func (ep *endpoint) rename(name string) error {
 	n.updateSvcRecord(ep, n.getController().getLocalEps(netWatch), false)
 
 	oldName := ep.name
+        oldAnonymous := ep.anonymous
```
Comment (thaJeztah) [inline endpoint.go]: looks like you're using spaces here for indentation; other lines use tabs; can you change this to tabs?

File: endpoint.go [hunk 2]
```diff
@@ -533,13 +533,16 @@ func (ep *endpoint) rename(name string) error {
...
 	ep.name = name
-
+        ep.anonymous = false
```
Comment (thaJeztah) [inline endpoint.go]: same here

**Comment by jmzwcn** [comment]
still failed on CircleCI?

**Comment by thaJeztah** [comment]
@jmzwcn it's complaining about `gofmt`, so it looks like your code needs a `gofmt`. you can try running

    gofmt -s -w endpoint.go

and adding/committing the changes. While you're doing this, perhaps you can also "squash" your commits https://docs.docker.com/opensource/workflow/work-issue/#pull-and-rebase-frequently

**Comment by jmzwcn** [comment]
@thaJeztah, yes, it works as you said. Thank you!

**Comment by sanimej** [comment]
@jmzwcn Thanks for working on this. This change will address the issue if the container is already running when the rename is done. But it won't fix the problem if the rename is done before starting the container.

````
docker create -it --net net1 busybox top
docker rename <id> c1
docker start c1

docker create -it --net net1 --name c2 busybox
````
c2 can't resolve c1's name. Its because the anonymous state of an interface is maintained on the daemon side and that has to be updated as well. Can you push a PR in docker to take care of that ?

**Comment by jmzwcn** [comment]
OK, I will handle un-running status in docker line.

**Comment by jmzwcn** [comment]
Have created another PR for it. https://github.com/docker/docker/pull/22716

**Comment by sanimej** [comment]
LGTM

**Pull request merged**